obs/project.go

```diff
@@ -40,13 +40,69 @@ type Project struct {
 	UseForBuild  *UseForBuild `json:"useForBuild,omitempty" xml:"useforbuild,omitempty"`
 }
 
-type Client struct {
-	Client   http.Client
+// OBS is a wrapper around OBS related functionality
+type OBS struct {
+	client  Client
+	options *Options
+}
+
+type obsClient struct {
+	*http.Client
+}
+
+// Client is an interface modeling supported OBS operations
+type Client interface {
+	InvokeOBSEndpoint(ctx context.Context, username string, password string, method string, url string, xml *bytes.Buffer) (*http.Response, error)
+}
+
+// InvokeOBSEndpoint invokes an OBS endpoint by making a HTTP request
+func (o *obsClient) InvokeOBSEndpoint(ctx context.Context, username string, password string, method string, url string, xmlData *bytes.Buffer) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, xmlData)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: creating request: %v", err),
+		}
+	}
+
+	req.SetBasicAuth(username, password)
+	req.Header.Set("Accept", "application/xml; charset=utf-8")
+
+	resp, err := o.Client.Do(req)
+	if err != nil {
+		return nil, &APIError{
+			HTTPStatusCode: 0,
+			OBSStatusCode:  "",
+			Message:        fmt.Sprintf("creating obs project: sending request: %v", err),
+		}
+	}
+
+	return resp, nil
+}
+
+// Options is a set of options to configure the behavior of the OBS package
+type Options struct {
 	Username string
 	Password string
 	APIURL   string
 }
 
+// DefaultOptions return an options struct with commonly used settings
+func DefaultOptions() *Options {
+	return &Options{
+		APIURL: "https://build.opensuse.org/",
```

Comment:
API URL is incorrect
```suggestion
		APIURL: "https://api.opensuse.org/",
```
from https://api.opensuse.org/apidocs/index